detox/src/devices/common/drivers/android/tools/HashHelper.js

```diff
@@ -0,0 +1,64 @@
+const crypto = require('crypto');
+const fs = require('fs');
+
+const { DetoxInternalError } = require('../../../../../errors');
+const path = require('path');
+
+const HASH_PATH = '/data/local/tmp/detox';
+
+class HashHelper {
```

Comment:
I'm not entirely against classes like these, but I really think it needs to be broken down into at least two separate pieces:
1. Calculating a hash: This is a concern that can be introduced as a **top-level** (non-Android specific) **util**. It could offer methods such as calculating hashes async'ly over strings, or - as a convenience, based on in-streams (as in this case...).
1. Saving the hash to the device: This is something that can indeed be under a helper as such. However, **it should utilize `TempFileXfer`** in order to send the file away: That would allow a reuse of:
    1. The target path, which is unnecessarily redefined here, at the moment.
    1. Creating the dest dir on the device, which isn't called here - but should be.

Reading the hash back from the device + comparing it can be done in the driver itself, I suppose.